Code/PgSQL/rdkit/rdkit.h

```diff
@@ -40,6 +40,11 @@ extern "C" {
 
 #include <postgres.h>
 
+#define RDKIT_FREE_IF_COPY_P(ptrsrc, ptrori)                   \
+  do {                                                         \
+    if ((Pointer)(ptrsrc) != (Pointer)(ptrori)) pfree(ptrsrc); \
```

Comment:
Stupid question: why is this in a `do { } while()` that's only going to run once?

Reply:
The do/while loop is just a syntactic trick that allows using the macro more easily in the code, without requiring a different handling of semicolons or curly braces depending on where it's placed. This blog post provides a better explanation and a few examples https://hownot2code.wordpress.com/2016/12/05/do-while-0-in-macros/ (I would have said this construct was already in use somewhere else in RDKit too, but I'm not finding it, I was probably mistaken).